base/array.jl

```diff
@@ -412,9 +453,21 @@ function getindex(::Type{T}, vals...) where T
     return a
 end
 
+# safe version
+function getindex(::Type{T}, vals::T...) where T
+    @inline
+    @_effect_free_terminates_locally_meta
+    a = Vector{T}(undef, length(vals))
+    @_safeindex for i in 1:length(vals)
+        a[i] = vals[i]
+    end
+    return a
+end
```

Comment:
@aviatesk
Is there a reason to add this? It seems unneeded and cause some new ambiguity https://github.com/JuliaArrays/StaticArrays.jl/pull/1153#issuecomment-1500790911

Reply:
I added this definition since otherwise we can't const-fold safe `getindex` operations due to the `:nothrow` effect tainted by the `foldr` branch in the unsafe definition.

Reply:
But it looks like this definition is no longer needed (the `vals isa NTuple` branch is const-folded even without constant information propagated)? I will make a PR to delete this.

Reply:
https://github.com/JuliaLang/julia/pull/49310